src/darwin/Framework/CHIP/MTRNOCChainIssuer.h

```diff
@@ -0,0 +1,48 @@
+/**
+ *
+ *    Copyright (c) 2022 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#import <Foundation/Foundation.h>
+
+#import <Matter/MTRAttestationInfo.h>
+#import <Matter/MTRCSRInfo.h>
+
+NS_ASSUME_NONNULL_BEGIN
+
+@protocol MTRNOCChainIssuer <NSObject>
+@required
+
+/**
+ * @brief When a MTRNOCChainIssuer is set for this controller, then onNOCChainGenerationNeeded will be
+ * called when the NOC CSR needs to be signed. This allows for custom credentials issuer
+ * implementations, for example, when a proprietary cloud API will perform the CSR signing.
+
+ * The commissioning workflow will stop upon the onNOCChainGenerationNeeded callback and
+ * resume once onNOCChainGeneration is called.
```

Comment:
Once onNOCChainGeneration is called on what?

Reply:
This is now passed as a block to the NOCChainIssuer. So, in the current revision, the docs don't need more context anymore.